packages/next/src/build/flying-shuttle/store-shuttle.ts

```diff
@@ -8,20 +12,59 @@ import {
   PAGES_MANIFEST,
   ROUTES_MANIFEST,
 } from '../../shared/lib/constants'
-import { recursiveCopy } from '../../lib/recursive-copy'
+import { getNextPublicEnvironmentVariables } from '../webpack/plugins/define-env-plugin'
+
+export function generateShuttleManifest(config: NextConfigComplete) {
+  // NEXT_PUBLIC_ changes for now since they are inlined
+  // and specific next config values that can impact the build
+  const globalHash = crypto.createHash('sha256')
+
+  for (const key in getNextPublicEnvironmentVariables()) {
+    if (key.startsWith('NEXT_PUBLIC_')) {
```

Comment:
this seems redundant with the refactor to use `getNextPublicEnvironmentVariables`